### DISCUSSION TITLE
Possibility to have both constructors and setters called

### DISCUSSION THREAD

**Discussion by magnetik** [CuyZ/Valinor#89] [Q&A] [answered]
Hi,

I've some object that have some properties defined in the constructor while others can be accessed through setter.

Is this something Valinor support ? 

Thanks again for the lib :)

**Comment by romm** [comment, marked as answer]
Hi @magnetik, this library recommends objects immutability so it wont allow automatic setters calls; however if for some reason you need this kind of operation, you can use the object binding like below (there is no documentation yet because I'm not sure the current implementation will make it to v1.0):

```php
(new \CuyZ\Valinor\MapperBuilder())
    ->bind(function (string $string, int $int, float $float): SomeClassWithSetter {
        $class = new SomeClassWithSetter($string, $int);
        $class->setFloat($float);

        return $class;
    })
    ->mapper()
    ->map(SomeClassWithSetter::class, [/* … */]);
```

**Reply by magnetik** [reply]
Awesome. I think I'll make a named constructor with all parameters to avoid this.

**Reply by romm** [reply]
This is indeed a far better approach IMO!
